registry/client/authchallenge.go

```diff
@@ -0,0 +1,150 @@
+package client
+
+import (
+	"net/http"
+	"strings"
+)
+
+// Octet types from RFC 2616.
+type octetType byte
+
+// AuthorizationChallenge carries information
+// from a WWW-Authenticate response header.
+type AuthorizationChallenge struct {
+	Scheme     string
+	Parameters map[string]string
+}
+
```

Comment:
Not sure why this is even exposed, I will trim down some as this shouldn't be needed outside this package.